src/pyodide/internal/workers.py

```diff
@@ -149,54 +157,208 @@ def json(
+        self._js_form_data.set(key, _py_value_to_js(value))
+
+    def append(self, key: str, value: FormDataValue, filename: str | None = None):
+        self._js_form_data.append(key, _py_value_to_js(value), filename)
 
     def delete(self, key: str):
-        self.js_form_data.delete(key)
+        self._js_form_data.delete(key)
 
     def __contains__(self, key: str) -> bool:
-        return self.js_form_data.has(key)
+        return self._js_form_data.has(key)
 
     def values(self) -> Generator[FormDataValue, None, None]:
-        yield from self.js_form_data.values()
+        for val in self._js_form_data.values():
+            yield _js_value_to_py(val)
 
     def keys(self) -> Generator[str, None, None]:
-        yield from self.js_form_data.keys()
+        yield from self._js_form_data.keys()
 
     def __iter__(self):
         yield from self.keys()
 
     def items(self) -> Generator[tuple[str, FormDataValue], None, None]:
-        for item in self.js_form_data.entries():
-            yield (item[0], item[1])
+        for k, v in self._js_form_data.entries():
+            yield (k, _js_value_to_py(v))
 
     def __delitem__(self, key: str):
         self.delete(key)
 
     def __len__(self):
         return len(self.keys())
+
+    def get_all(self, key: str) -> list[FormDataValue]:
+        return list(map(lambda x: _js_value_to_py(x), self._js_form_data.getAll(key)))
+
+    @property
+    def js_object(self) -> "js.FormData":
+        return self._js_form_data
+
+
+def _supports_buffer_protocol(o):
+    try:
+        # memoryview used only for testing type; 'with' releases the view instantly
+        with memoryview(o):
+            return True
+    except TypeError:
+        return False
+
+
+@contextmanager
+def _make_blob_entry(e):
+    if isinstance(e, str):
+        yield e
+        return
+    if isinstance(e, Blob):
+        yield e._js_blob
+        return
+    if hasattr(e, "constructor") and (e.constructor.name in ("Blob", "File")):
+        yield e
+        return
+    if _supports_buffer_protocol(e):
+        px = create_proxy(e)
+        buf = px.getBuffer()
+        try:
+            yield buf.data
+            return
+        finally:
+            buf.release()
+            px.destroy()
+    raise TypeError(f"Don't know how to handle {type(e)} for Blob()")
+
+
+BlobValue = (
+    "str | bytes | js.ArrayBuffer | js.TypedArray | js.DataView | js.Blob | Blob"
+)
+
+
+class BlobEnding(StrEnum):
+    TRANSPARENT = "transparent"
+    NATIVE = "native"
+
+
+class BlobKwargs(TypedDict, total=False):
+    content_type: str
+    endings: BlobEnding
+
+
+class Blob:
+    def __init__(
+        self,
+        blob_parts: "Iterable[BlobValue] | BlobValue",
+        **options: Unpack[BlobKwargs],
+    ):
+        if "content_type" in options:
+            options["type"] = options["content_type"]
+        if "endings" in options:
+            options["endings"] = str(options["endings"])
```

Comment:
Seems like it might be better to be explicit here:
```suggestion
        content_type: str | None = None,
        endings: BlobEnding | str | None = None
    ):
        if endings:
           endings = str(endings)
```
and then:
```py
    js.Blob.new(args, type=content_type, endings=endings)
```
Also, I am pretty sure that you don't need `to_js` for `args`.

Reply:
> Also, I am pretty sure that you don't need to_js for args.

You do, I get this otherwise: `pyodide.ffi.JsException: TypeError: Failed to construct 'Blob': constructor parameter 1 is not of type 'Array'.`

Reply:
That's unfortunate. There's been discussion before whether we should make `Array.isArray()` return `true` on a Python list. Not sure whether it would fix this error message. I also don't remember why I didn't do it -- it's possible I just never got to it, or that there was some problem it caused.